nixos/modules/system/boot/loader/generic-extlinux-compatible/default.nix

```diff
@@ -65,6 +65,24 @@ in
         '';
       };
 
+      copyKernels = mkOption {
+        default = false;
+        type = types.bool;
+        description = ''
+          Whether to copy the necessary boot files into /boot, so
+          /nix/store is not needed by the boot loader. This is done
+          automatically if /boot is on a different partition than /.
```

Comment:
You seem to have copied the description from `boot.loader.grub.copyKernels` but your code behaves differently.